File: hello-sine.cpp
/*
Feito por Hikari :)
*/
#include <iostream>
#include <math.h>
#include <string>

//return a string that repeat itseld >multiplier< times
std::string repeat(std::string raw_string, int multiplier)
{
	std::string final_string="";
	
	for(int i=0; i<multiplier; i++)
	{
		final_string = final_string+raw_string;
	}
	
	return final_string;
}

int main()
{
	//declare angle and sine value variable
	float angle=0;
	int sine;
	
	//declare modifiers
	float in_multiplier=1;
	float out_multiplier=40;
	float add=40;
	
	//gets string read for printing	
	std::string string = "I know C++ :) ";
	string = repeat(string, out_multiplier+add);
	
	//loop
	while(1)
	{
		//increase angle
		angle+=0.005;
		
		//calculate sine
		sine= sin(angle*in_multiplier)*out_multiplier+add;
		
		//print
		std::cout<<string.substr(0,sine)<<"\n";
	}
	
	return 0;
}

File: hello-sine.rb
"""
Feito pela Hikari :)
"""

# declares angle which will increase after each interation of the loop
angle = 0
string_raw = 'I know ruby :)'

# modifyers to the function
in_multiplier = 5/10.0
out_multiplier = 40
add = 40

# loop
while true do
    # increases angle
    angle += 0.1

    # calculates sine
    sine = Math.sin(angle*in_multiplier)*out_multiplier+add

    # format string
    string_format = string_raw * (sine/string_raw.length+1).to_i

    # print the right amount of characters
    p string_format[...sine.to_i]

    sleep(0.01)
end




